docs/android/v2/guides/token-endpoint.mdx

```diff
@@ -0,0 +1,74 @@
+---
+title: Auth Token Endpoint Guide
+nav: 2.3
+---
+
+## Get an app token using token endpoint
```

Comment:
needs an mention of 
- sample apps use this + reason on why do they use this
- link to app token foundation
   - along with this, need a mention that it's required for join room in the client sdk